cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -176,6 +184,27 @@ protected void startUp() throws Exception {
                                           AppMetadataStore.create(context).getApplicationCount());
     metricsCollectionService.getContext(Collections.emptyMap()).gauge(Constants.Metrics.Program.APPLICATION_COUNT,
                                                                       count);
+
+    Map<ImmutablePair<String, String>, Long> pluginCounts = TransactionRunners.run(transactionRunner,
+                                                            (TxCallable<Map<ImmutablePair<String, String>, Long>>)
+                                                              context ->
+                                                              getPluginCounts(AppMetadataStore.create(context)));
+    for (Map.Entry<ImmutablePair<String, String>, Long> entry : pluginCounts.entrySet()) {
+      Map<String, String> tags = ImmutableMap.of(Constants.Metrics.Tag.PLUGIN_NAME, entry.getKey().getFirst(),
+                                                 Constants.Metrics.Tag.PLUGIN_TYPE, entry.getKey().getSecond());
+
+      metricsCollectionService.getContext(tags).gauge(Constants.Metrics.Program.PLUGIN_COUNT, entry.getValue());
+    }
+  }
+
+  private Map<ImmutablePair<String, String>, Long> getPluginCounts(AppMetadataStore appMetadataStore)
+    throws IOException {
+    return appMetadataStore.getAllApplications().stream().map(ApplicationMeta::getSpec)
+        .map(ApplicationSpecification::getPlugins)
+        .flatMap(e -> e.values().stream())
+        .map(Plugin::getPluginClass)
+        .map(e -> ImmutablePair.of(e.getName(), e.getType()))
+      .collect(Collectors.groupingByConcurrent(Function.identity(), Collectors.counting()));
```

Comment:
Misalignment.

Also, why you need a concurrent map as the result?

Reply:
yeah concurrent map is not needed; removed. also fixed alignment.